tests/nodes/subobjects/test_software.py

```diff
@@ -21,3 +22,18 @@ def test_setter_getter(complex_software_node):
     assert s2.version == "v0.3.0"
     s2.source = "https://github.com/SSAGESLabs/PySAGES"
     assert s2.source == "https://github.com/SSAGESLabs/PySAGES"
+
+
+def test_uuid(complex_software_node):
+    s = complex_software_node
+
+    # Deep copies should not share uuid (or uids) or urls
+    s2 = copy.deepcopy(complex_software_node)
```

Comment:
can we use the regular `complex_software_node` here without `deepcopy` ?

Reply:
No, if you use the `complex_software_node` directly, they have the same uuid.
They are even the same python instance.
So there is no point in checking if their uuid id changed as expected.
The deepcopy is absolutely critical to this here.

Reply:
okay got it!